pkg/controller/node/cidr_set.go

```diff
@@ -38,10 +38,16 @@ type cidrSet struct {
 func newCIDRSet(clusterCIDR *net.IPNet, subNetMaskSize int) *cidrSet {
 	clusterMask := clusterCIDR.Mask
 	clusterMaskSize, _ := clusterMask.Size()
-	maxCIDRs := 1 << uint32(subNetMaskSize-clusterMaskSize)
+
+	var maxCIDRs int
```

Comment:
That's fine, add a todo / open an issue that sparse ranges should be better supported.  There are enough sparse bitfield compression algos that we could have much larger ranges at some point in the future.